apps/mosaico/src/Mosaico/Frontpage.purs

```diff
@@ -10,105 +13,109 @@ import Data.Newtype (un)
 import Data.String (contains)
 import Data.String.Pattern (Pattern(..))
 import Data.Tuple (Tuple(..))
-import Data.Tuple.Nested ((/\))
 import Foreign.Object as Object
 import KSF.HtmlRenderer (render) as HtmlRenderer
-import KSF.HtmlRenderer.Models (getAttribs, getName, replacingHook) as HtmlRenderer
 import KSF.Spinner (loadingSpinner)
 import Lettera.Models (ArticleStub, Tag(..), tagToURIComponent)
-import Mosaico.Models (ArticleFeed(..))
+import Mosaico.Frontpage.Models (Hook, toHookRep)
 import React.Basic (JSX)
 import React.Basic.DOM as DOM
 import React.Basic.Events (EventHandler)
 
-type Props =
-  { content :: Maybe ArticleFeed
+-- | Represents a frontpage
+class Frontpage h where
```

Comment:
I don't think type class is the right choice for this. I would find it more idiomatic to replace both
```
newtype ListFrontpage = ListFrontpage ListFrontpageProps
newtype PrerenderedFrontpage = PrerenderedFrontpage PrerenderedFrontpageProps
```

with
```
data Frontpage = ListFrontpage ListFrontpageProps | PrerenderedFrontpage PrerenderedFrontpageProps
```

And replace
```
renderListFrontpage :: ListFrontpageProps -> JSX
renderPrerenderedFrontpage :: PrerenderedFrontpageProps -> JSX
```

with
```
render :: Frontpage -> JSX
```

with pattern matching for `ListFrontpage` and `PrerenderedFrontpage` to give the corresponding render implementation.

Reply:
I was thinking the same